src/headerssync.h

```diff
@@ -247,6 +255,13 @@ class HeadersSyncState {
     /** Height of m_presync_last_header_received */
     int64_t m_presync_height{0};
 
+    /** During phase 1 (PRESYNC), we cache received headers so we don't have to
+     *  re-download all of them in phase 2 (REDOWNLOAD). */
+    std::vector<CompressedHeader> m_headers_cache;
```

Comment:
why "cache" and not just `m_headers`? The comment might want to explain what happens to the headers that don't fit here (I assume the key is the height, being the vector's index).